sklearn/neighbors/_classification.py

```diff
@@ -281,6 +282,14 @@ def predict(self, X):
         n_outputs = len(classes_)
         n_queries = _num_samples(X)
         weights = _get_weights(neigh_dist, self.weights)
+        if weights is not None:
+            weights = check_array(weights, dtype=[np.float32, np.float64])
```

Comment:
I am wondering if can just keep `dtype="numeric"` by making the Cython function more flexible.
I assume that this can happen with `weights` is passed as a callable.